airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -566,6 +568,17 @@ def transform_function(original_value: Any, field_schema: Dict[str, Any]) -> Any
 
         return transform_function
 
+    # csv header field names for this report differ per marketplace (are localized to marketplace language)
+    # but columns come in the same order
+    # so we set fieldnames to our custom ones
+    # and skip the original header with next()
+    @staticmethod
+    def parse_document(document):
+        reader = csv.DictReader(StringIO(document), delimiter="\t", fieldnames=SellerFeedbackReports.NORMALIZED_FIELD_NAMES)
```

Comment:
This is a working workaround but not 100% safe if Amazon changes the structure of the CSV file. And I'm not sure that this API has versioning that could allow us to avoid these breaking changes.

Do you think you could raise an error if the length of `NORMALIZED_FIELD_NAMES` is different from the number of columns in the CSV file?
Do you have access to all the localized values? It could make the normalization safer by applying a mapping between localized and normalized values.

Reply:
Yes with amazon seller partner apis there is no certainty at all :D

Agreed with error raising on column count mismatch.

But mapping values is almost impossible, values are random, are not guaranteed to be the same tomorrow and I do not have access to all possible marketplaces to get them anyway.

Reply:
@alafanechere added header field count error raising on count mismatch